File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.c
int maxProduct(int n) {
    int a[10];
    int i = 0;
    while( n!= 0)
    {
        a[i] = n % 10;
        i++;
        n /= 10;
    }
    int maxProduct = INT_MIN;
    int size = sizeof(a) / sizeof(a[0]);
    for(int i=0; i<size; i++)
    {
        for(int j=i+1; j<size; j++)
        {
            if(a[i] * a[j] > maxProduct)
                maxProduct = a[i] * a[j];
        }
    }
    return maxProduct;
}

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        largest = second_largest = 0
        while n > 0:
            n, digit = divmod(n, 10)
            if digit > largest:
                second_largest, largest = largest, digit
            elif digit > second_largest:
                second_largest = digit
        return largest * second_largest